src/Control/Monad/Catch.hs

```diff
@@ -178,14 +199,13 @@ class MonadCatch m => MonadMask m where
   -- and/or unkillable.
   uninterruptibleMask :: ((forall a. m a -> m a) -> m b) -> m b
 
-  -- | A generalized version of the standard bracket function which allows
-  -- distinguishing different exit cases. Instead of providing it a single
-  -- release action, this function takes two different actions: one for the
-  -- case of a successful run of the inner function, and one in the case of an
-  -- exception. The former function is provided the acquired value, while
-  -- the exception release function is provided both the acquired value and
-  -- the exception that was thrown. The result values of both of these
-  -- functions are ignored.
+  -- | A generalized version of 'bracket' which uses 'ExitCase' to distinguish
+  -- the different exit cases, and returns the values of both the 'use' and
+  -- 'release' actions. You should probably use 'bracket' instead.
```

Comment:
I'd probably clarify _why_ `bracket` is better to use (more ergonomic), and comment that many other functions (like `finally`) are built on top of this method.